### DISCUSSION TITLE
Regex: How to get the value of a capture group when the input is a ReadOnlySpan<char>?

### DISCUSSION THREAD

**Discussion by elgonzo** [dotnet/runtime#107510] [Q&A] [answered]
I struggle with the `ReadOnlySpan<char>`-based API for `System.Text.RegularExpressions.Regex`.

To illustrate my problem, lets imagine this simple task: For the given input string:
```C#
ReadOnlySpan<char> input = "12a34b567c890a123c45b678c90";
```
get the integer values in `a(\d+)b(\d+)c` and ignore the rest. (The example input is a sequence of numbers delimited by specific word identifiers, with the identifiers represented by a single letter here to keep the example simple and readable.)

With the `string`-based `Regex.Match` API that's returning a `Match` instance, solving the task is a walk in the park. But i don't have a `string`. I have a `ReadOnlySpan<char>`. And `Regex.EnumerateMatches` isn't up for the task -- it only gives me the start index and length of the match for the entire `a(\d+)b(\d)c` pattern but not for the capture groups.

So, now what? Is the only way forward (with respect to using Regex) to turn the `ReadOnlySpan<char>` into a string, or is there a better solution hidden from my eyes?

<br>

(P.S.: I am not interested in workarounds like employing a sequence of multiple regex pattern matches that break down the input into the parts of interest, and/or copious amounts of look-ahead and look-behind assertions. Because such an approach becomes quickly impractical and error-prone not only from a code maintenance view but especially if the regex patterns need to be configurable outside of the code (for example through a configuration/INI file).

**Comment by stephentoub** [comment, marked as answer]
> turn the ReadOnlySpan<char> into a string

Yes. Span-based input that also supports full reporting of captures is covered by https://github.com/dotnet/runtime/issues/73223.
